build_runner/lib/src/asset/reader.dart

```diff
@@ -3,37 +3,63 @@ import 'dart:convert';
 
 import 'package:build/build.dart';
 import 'package:glob/glob.dart';
+import '../asset_graph/graph.dart';
+import '../asset_graph/node.dart';
 
 abstract class RunnerAssetReader extends AssetReader {
+  @override
   Iterable<AssetId> findAssets(Glob glob, {String packageName});
 
   /// Asynchronously gets the last modified [DateTime] of [id].
   Future<DateTime> lastModified(AssetId id);
 }
 
-class AssetReaderSpy implements AssetReader {
+final _false = new Future.value(false);
```

Comment:
Renamed to _asyncFalse. We can change to `FutureOr<bool>` in build 0.9.0 and clean this up.
Unfortunately it is a breaking change in case anyone does `reader.hasInput(input).then()`